cache/network/upload.go

```diff
@@ -0,0 +1,70 @@
+package network
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/retryhttp"
+)
+
+// UploadParams ...
+type UploadParams struct {
+	APIBaseURL  string
+	Token       string
+	ArchivePath string
+	ArchiveSize int64
+	CacheKey    string
+}
+
+// Upload a cache archive and associate it with the provided cache key
+func Upload(params UploadParams, logger log.Logger) error {
+	validatedKey, err := validateKey(params.CacheKey)
+	if err != nil {
+		return err
+	}
+
+	client := newAPIClient(retryhttp.NewClient(logger), params.APIBaseURL, params.Token)
+
+	logger.Debugf("Get upload URL")
+	prepareUploadRequest := prepareUploadRequest{
+		CacheKey:           validatedKey,
+		ArchiveFileName:    filepath.Base(params.ArchivePath),
+		ArchiveContentType: "application/zstd",
+		ArchiveSizeInBytes: params.ArchiveSize,
+	}
+	resp, err := client.prepareUpload(prepareUploadRequest)
+	if err != nil {
+		return fmt.Errorf("failed to get upload URL: %w", err)
+	}
+	logger.Debugf("Upload ID: %s", resp.ID)
+
+	logger.Println()
```

Comment:
I would either remove it or make it a debug log

Reply:
Fixed